solana-programs/staking-bridge/tests/staking-bridge.ts

```diff
@@ -1,16 +1,293 @@
 
-describe("staking-bridge", () => {
-  // Configure the client to use the local cluster.
-  anchor.setProvider(anchor.AnchorProvider.env());
+const {
+  Connection,
+  Keypair,
+  PublicKey,
+  SystemProgram,
+  SYSVAR_CLOCK_PUBKEY,
+  SYSVAR_RENT_PUBKEY,
+} = anchor.web3
+
+// Read in fee payer from the environment variable
+const FEE_PAYER_SECRET = process.env.feePayerSecret
+const feePayerKeypair = Keypair.fromSecretKey(
+  Uint8Array.from(JSON.parse(FEE_PAYER_SECRET))
+)
+const feePayerPublicKey = feePayerKeypair.publicKey
+
+const SOL_USDC_TOKEN_ADDRESS_KEY = new PublicKey(SOL_USDC_TOKEN_ADDRESS)
+const SOL_AUDIO_TOKEN_ADDRESS_KEY = new PublicKey(SOL_AUDIO_TOKEN_ADDRESS)
+
+const endpoint = 'https://api.mainnet-beta.solana.com'
+const connection = new Connection(endpoint, 'confirmed');
+
+async function signTransaction(transaction) {
+  transaction.partialSign(feePayerKeypair)
+  return transaction
+}
+async function signAllTransactions(transactions: any[]) {
+  transactions.forEach((transaction) => {
+    transaction.partialSign(feePayerKeypair)
+  })
+  return transactions
+}
+
+describe('staking-bridge', () => {
+  const wallet = {
+    publicKey: feePayerPublicKey,
+    signTransaction,
+    signAllTransactions
+  }
+  const provider = new anchor.AnchorProvider(connection, wallet, { skipPreflight: true });
+  anchor.setProvider(provider)
 
   const program = anchor.workspace.StakingBridge as Program<StakingBridge>;
 
-  it("Is initialized!", async () => {
+  const [stakingBridgePda, stakingBridgePdaBump] = PublicKey.findProgramAddressSync(
+    [Buffer.from('staking_bridge')],
+    program.programId
+  )
+
+  it('creates the staking bridge pda', async () => {
+    // Add your test here.
+    const tx = await program.methods
+      .createPda()
+      .accounts({
+        stakingBridgePda,
+        payer: feePayerPublicKey,
+        systemProgram: SystemProgram.programId,
+      })
+      .rpc();
+    console.log('Your transaction signature', tx);
+  })
+
+  it('swaps SOL USDC to SOL AUDIO', async () => {
+    const market = await getMarket(connection, serumMarketPublicKey.toString(), serumDexProgram.toString())
+    console.log("serum market info:", JSON.stringify(market))
+
+    const poolKeys = await getAssociatedPoolKeys({
+      programId: ammProgram,
+      serumProgramId: serumDexProgram,
+      marketId: market.address,
+      baseMint: market.baseMint,
+      quoteMint: market.quoteMint
+    })
+    console.log("amm poolKeys: ", JSON.stringify(poolKeys))
+
+    const { vaultOwner, vaultNonce } = await getVaultOwnerAndNonce(serumMarketPublicKey, serumDexProgram)
+    if (vaultNonce.toNumber() != market.vaultSignerNonce) {
+      console.log(
+        "withdraw vaultOwner:",
+        vaultOwner.toString(),
+        "vaultNonce: ",
+        vaultNonce.toNumber(),
+        "market.vaultSignerNonce:",
+        market.vaultSignerNonce.toString()
+      )
+      throw ("vaultSignerNonce incorrect!")
```

Comment:
instead of throw, this should be an assert?

Reply:
copied this and forgot to change. will handle